Geometry/GEMGeometryBuilder/src/GEMGeometryBuilderFromCondDB.cc

```diff
@@ -76,65 +80,108 @@ GEMGeometry* GEMGeometryBuilderFromCondDB::build(const RecoIdealGeometry& rgeo)
       
-      //Change of axes for the forward
-    Basic3DVector<float> newX(1.,0.,0.);
-    Basic3DVector<float> newY(0.,0.,1.);
+    //Change of axes for the forward
+    Basic3DVector<float> newX( 1., 0., 0. );
+    Basic3DVector<float> newY( 0., 0., 1. );
     //      if (tran[2] > 0. )
     newY *= -1;
-    Basic3DVector<float> newZ(0.,1.,0.);
-    rot.rotateAxes (newX, newY, newZ);	
+    Basic3DVector<float> newZ( 0., 1., 0. );
+    rot.rotateAxes( newX, newY, newZ );	
   
-    
-    BoundPlane* bp = new BoundPlane(pos, rot, bounds);
-    ReferenceCountingPointer<BoundPlane> surf(bp);
-    GEMEtaPartition* gep=new GEMEtaPartition(gemid, surf, e_p_specs);
+    BoundPlane* bp = new BoundPlane( pos, rot, bounds );
+    ReferenceCountingPointer<BoundPlane> surf( bp );
+    GEMEtaPartition* gep = new GEMEtaPartition( gemid, surf, epSpecs );
+    LogDebug("GEMGeometryBuilder") << "GEM Eta Partition created with id = " << gemid
+				   << " and added to the GEMGeometry" << std::endl;
     geometry->add(gep);
     
-    
-    std::list<GEMEtaPartition *> gepls;
-    /*
-    if (chids.find(chid)!=chids.end()){
-      gepls = chids[chid];
+    std::list<GEMEtaPartition *> gepList;
+    if( m_chids.find( chid ) != m_chids.end()) {
+      gepList = m_chids[chid];
     }
-    */
-    gepls.emplace_back(gep);
-    //chids[chid]=gepls;
-    
+    gepList.emplace_back(gep);
+    m_chids[chid] = gepList;
   }
-  /*
+  
   // Create the GEMChambers and store them on the Geometry 
 
-  for( std::map<GEMDetId, std::list<GEMEtaPartition *> >::iterator ich=chids.begin();
-       ich != chids.end(); ich++){
-    GEMDetId chid = ich->first;
-    std::list<GEMEtaPartition * > gepls = ich->second;
+  for( auto& ich : m_chids ) {
+    GEMDetId chid = ich.first;
+    std::list<GEMEtaPartition * > gepList = ich.second;
 
     // compute the overall boundplane. At the moment we use just the last
     // surface
-    BoundPlane* bp=0;
-    for(std::list<GEMEtaPartition *>::iterator gepl=gepls.begin();
-    gepl!=gepls.end(); gepl++){
-    const BoundPlane& bps = (*gepl)->surface();
-      bp = const_cast<BoundPlane *>(&bps);
+    BoundPlane* bp = 0;
+    for( auto& gep : gepList ) {
+      const BoundPlane& bps = ( *gep ).surface();
+      bp = const_cast<BoundPlane *>( &bps );
     }
 
-    ReferenceCountingPointer<BoundPlane> surf(bp);
+    ReferenceCountingPointer<BoundPlane> surf( bp );
+    // Create the superchamber
+    if( chid.layer() == 1 ) {
+      GEMDetId schid( chid.region(), chid.ring(), chid.station(), 0, chid.chamber(), 0 );
+      GEMSuperChamber* sch = new GEMSuperChamber( schid, surf );
+      LogDebug("GEMGeometryBuilder") << "GEM SuperChamber created with id = " << schid
+				     << " and added to the GEMGeometry" << std::endl;
+      superChambers.emplace_back( sch );
+    }
+    
     // Create the chamber 
-    GEMChamber* ch = new GEMChamber (chid, surf); 
+    GEMChamber* ch = new GEMChamber( chid, surf ); 
```

Comment:
@jshlee - please, consider using unique or shared pointers

Reply:
for all new objects or just this one?

Reply:
for all, please

Reply:
@ianna did you decide this comment can be ignored?

Reply:
@kpedro88 - I'm going to fix all geometry/DD code to use smart pointers, so yes, it's ok to leave some here :-)